core/flow/rule_manager.go

```diff
@@ -3,42 +3,154 @@ package flow
 import (
 	"encoding/json"
 	"fmt"
+	"sync"
+
 	"github.com/alibaba/sentinel-golang/logging"
 	"github.com/pkg/errors"
-	"sync"
 )
 
-// const
 var (
 	logger = logging.GetDefaultLogger()
 )
 
+// SafeTrafficControllers represents the threadSafe map storage for TrafficShapingController.
+type SafeTrafficControllers struct {
+	controller map[string][]*TrafficShapingController
+	mux        sync.Mutex
```

Comment:
RWMutex is better.
Here is a typical scenario of reading too much and writing too little.

Reply:
ok, will use RWMutex